internal/graph/dispatch_throttling_check_resolver_test.go

```diff
@@ -91,13 +173,72 @@ func TestDispatchThrottlingCheckResolver(t *testing.T) {
 
 		ctx := context.Background()
 
+		var resolveCheckErr error
+
+		go func() {
+			defer goFuncDone.Done()
+			goFuncInitiated.Done()
+			_, resolveCheckErr = dut.ResolveCheck(ctx, req)
+		}()
+
+		goFuncInitiated.Wait()
+		// Before we send the tick, we don't expect the dispatch to run
+		time.Sleep(1 * time.Millisecond)
```

Comment:
This is superfulous, because it doesn't guarantee that `ResolveCheck` has been called before this assertion anyways (`goFuncInitiated.Done()` runs before `ResolveCheck` is invoked).

At the time you run the assertion you shouldn't expect the dispatch to have run yet, because we haven't produced a tick. So can't we just drop the `goFuncInitiated` WaitGroup and run this assertion right after the goroutine and expect the same outcome?

Only after the `nonBlockingSend` shouold we expect a mutation to the resolver counter.

Reply:
We still need to have goFuncInitiated.  Otherwise, we may have signal before the item is in queue.  Thus, the signal is ignored (remember that it is no blocking send).

Reply:
Discussed offline - will use ticker to move forward